src/darwin/Framework/CHIP/MTRDevice_Concrete.mm

```diff
@@ -1860,25 +1870,51 @@ - (void)_handleUnsolicitedMessageFromPublisher
     [self _reattemptSubscriptionNowIfNeededWithReason:@"got unsolicited message from publisher"];
 }
 
-- (void)_markDeviceAsUnreachableIfNeverSubscribed
+- (void)_markDeviceAsUnreachableIfNeeded
 {
+    std::lock_guard lock(_lock);
+
+    // Nothing to do if subscription has established
+    if (HaveSubscriptionEstablishedRightNow(self->_internalDeviceState)) {
+        return;
+    }
+
+    // If subcription is actively being established, check later
+    if (self.lastSubscriptionActiveTime) {
+        NSTimeInterval intervalSinceDeviceLastActive = -[self.lastSubscriptionActiveTime timeIntervalSinceNow];
+        if (intervalSinceDeviceLastActive < MTR_DEVICE_ACTIVE_COMMUNICATION_THRESHOLD_SECONDS) {
+            // schedule 1/2 the active threshold in the future to check
+            [self _scheduleMarkDeviceAsUnreachableAfterWait:(MTR_DEVICE_ACTIVE_COMMUNICATION_THRESHOLD_SECONDS / 2)];
+            return;
+        }
+    }
+
+    // Use matter queue to avoid race with a timely report
```

Comment:
I'm not quite following this comment.  We have already decided to mark unreachable...  How will doing the remaining work off the Matter queue help with a race here?

Seems like we should maybe do all the above logic off the Matter queue, if we are trying to avoid races and lastSubscriptionActiveTime is set on that queue?

This is why we did one final check for HadSubscriptionEstablishedOnce on the Matter queue, to avoid stomping on the state.